packages/kit/src/runtime/load.js

```diff
@@ -5,9 +5,156 @@
 export function normalize(loaded) {
 	// TODO should this behaviour be dev-only?
 
-	if (loaded.error) {
-		const error = typeof loaded.error === 'string' ? new Error(loaded.error) : loaded.error;
+	if (
+		loaded.error ||
+		(loaded.status && loaded.status >= 400 && loaded.status <= 599 && !loaded.redirect)
+	) {
 		const status = loaded.status;
+		if (!loaded.error) {
+			let errorMessage = 'not assigned';
+			switch (status) {
+				case 400:
+					errorMessage = 'Bad Request';
+					break;
+				case 401:
+					errorMessage = 'Unauthorized';
+					break;
+				case 402:
+					errorMessage = 'Payment Required';
+					break;
+				case 403:
+					errorMessage = 'Forbidden';
+					break;
+				case 404:
+					errorMessage = 'Not Found';
+					break;
+				case 405:
+					errorMessage = 'Method not Allowed';
+					break;
+				case 406:
+					errorMessage = 'Not Acceptable';
+					break;
+				case 407:
+					errorMessage = 'Proxy Authentication Required';
+					break;
+				case 408:
+					errorMessage = 'Request Timeout';
+					break;
+				case 409:
+					errorMessage = 'Conflict';
+					break;
+				case 410:
+					errorMessage = 'Gone';
+					break;
+				case 411:
+					errorMessage = 'Length Required';
+					break;
+				case 412:
+					errorMessage = 'Precondition Failed';
+					break;
+				case 413:
+					errorMessage = 'Request Entity Too Large';
+					break;
+				case 414:
+					errorMessage = 'Request-URI Too Long';
+					break;
+				case 415:
+					errorMessage = 'Unsupported Media Type';
+					break;
+				case 416:
+					errorMessage = 'Requested Range Not Satisfiable';
+					break;
+				case 417:
+					errorMessage = 'Expectation Failed';
+					break;
+				case 418:
+					errorMessage = "I'm a teapot";
+					break;
+				case 422:
+					errorMessage = 'Unprocessable Entity';
+					break;
+				case 423:
+					errorMessage = 'Locked';
+					break;
+				case 424:
+					errorMessage = 'Failed Dependency';
+					break;
+				case 426:
+					errorMessage = 'Upgrade Required';
+					break;
+				case 428:
+					errorMessage = 'Precondition Required';
+					break;
+				case 429:
+					errorMessage = 'Too Many Requests';
+					break;
+				case 431:
+					errorMessage = 'Request Header Fields Too Large';
+					break;
+				case 444:
+					errorMessage = 'No Response';
+					break;
+				case 449:
```

Comment:
Some of the codes listed here such as this one are non-standard.

It actually seems invalid to send this `errorMessage`. It would need to include another header and is not a complete message
https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-wdv/1c1f4d5b-4fed-48ef-9bb6-893c2b25b74d
http://getstatuscode.com/449